tests/func/metrics/test_diff.py

```diff
@@ -3,11 +3,10 @@
 import yaml
 
 
-def test_metrics_diff_simple(tmp_dir, scm, dvc):
+def test_metrics_diff_simple(tmp_dir, scm, dvc, run_copy_metrics):
     def _gen(val):
-        tmp_dir.gen({"m.yaml": str(val)})
-        dvc.run(cmd="", metrics=["m.yaml"], single_stage=True)
-        dvc.scm.add(["m.yaml.dvc"])
+        tmp_dir.gen({"m_temp.yaml": str(val)})
+        run_copy_metrics("m_temp.yaml", "m.yaml", metrics=["m.yaml"])
```

Comment:
Similar code exists on metrics/params. A temp file with suffix `_temp/_t` is created which is copied (to have some `cmd` and generate metrics/plots as in the real usage).